libcperciva/util/sock_util.c

```diff
@@ -335,3 +336,36 @@ sock_addr_ensure_port(const char * addr)
 	/* Failure! */
 	return (NULL);
 }
+
+/**
+ * sock_addr_validate(addr):
+ * Check that ${addr} is syntactically valid, but do not perform any address
+ * resolution.
+ */
+int
+sock_addr_validate(const char * addr)
+{
+
+	/* Sanity checks. */
+	if (addr == NULL) {
+		warn0("Empty sock address.");
```

Comment:
s/sock/socket/; we use `sock` in the code for conciseness, but `socket` and `socket address` should be used in user-facing places.